tests/backends/hardware/sr/v4/test_power_board.py

```diff
@@ -0,0 +1,470 @@
+
+
+def test_cmd_read_button():
+    """Test that the CMD_READ_BUTTON."""
+    assert type(CMD_READ_BUTTON) is ReadCommand
+    assert CMD_READ_BUTTON.code == 8
+    assert CMD_READ_BUTTON.data_len == 4
+
+
+def test_cmd_read_fwver():
+    """Test that the CMD_READ_FWVER."""
+    assert type(CMD_READ_FWVER) is ReadCommand
+    assert CMD_READ_FWVER.code == 9
+    assert CMD_READ_FWVER.data_len == 4
+
+
+def test_cmd_write_output():
+    """Test that CMD_WRITE_OUTPUT works."""
+    assert len(CMD_WRITE_OUTPUT) == 6
+
+    for pos in PowerOutputPosition:
+        assert pos.value in CMD_WRITE_OUTPUT
+        command = CMD_WRITE_OUTPUT[pos.value]
+        assert pos.value == command.code
+
+
+def test_cmd_write_runled():
+    """Test that the CMD_WRITE_RUNLED."""
+    assert type(CMD_WRITE_RUNLED) is WriteCommand
+    assert CMD_WRITE_RUNLED.code == 6
+
+
+def test_cmd_write_errorled():
+    """Test that the CMD_WRITE_ERRORLED."""
+    assert type(CMD_WRITE_ERRORLED) is WriteCommand
+    assert CMD_WRITE_ERRORLED.code == 7
+
+
+def test_cmd_write_piezo():
+    """Test that the CMD_WRITE_PIEZO."""
+    assert type(CMD_WRITE_PIEZO) is WriteCommand
+    assert CMD_WRITE_PIEZO.code == 8
+
+
+def test_usb_communication_error():
+    """Test that USBCommunicationError works."""
+    u = USBCommunicationError(usb.core.USBError("Test."))
+    assert str(u) == "Test."
+    assert issubclass(USBCommunicationError, Exception)
+
+
+def test_usb_error_handler_decorator():
+    """Test that the handle_usb_error decorator works."""
+    @handle_usb_error
+    def test_func():
+        raise usb.core.USBError("Test")
+
+    with pytest.raises(USBCommunicationError):
+        test_func()
+
+
+class MockBoard(Board):
+    """
+    Never called.
+
+    To be removed after refactor. See #116.
+    """
+
+    @property
+    def name(self) -> str:
+        """Name of the board."""
+        pass
+
+    @property
+    def serial(self) -> str:
+        """The serial number."""
+        pass
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """Firmware version."""
+        pass
+
+    def make_safe(self) -> None:
+        """Make everything safe."""
+        pass
+
+    @staticmethod
+    def supported_components() -> List[Type["Component"]]:
+        """The supported components."""
+        pass
+
+    @staticmethod
+    def discover(backend: Backend) -> List["Board"]:
+        """Discover boards."""
+        pass
+
+
+class MockUSBContext:
+    """This class mocks the behaviour of usb.core.Context."""
+
+    def dispose(self, device: "MockUSBPowerBoardDevice"):
+        """Dispose of the device."""
+        pass
+
+
+class MockUSBPowerBoardDevice:
+    """This class mocks the behaviour of a USB device for a Power Board."""
+
+    def __init__(self, serial_number: str, fw_version: int = 3):
+        self.serial = serial_number
+        self.firmware_version = fw_version
+        self._ctx = MockUSBContext()
```

Comment:
`self._ctx` is unused (looks like a remnant from `usb1`). The `MockUSBContext` class can probably be deleted too.

Reply:
It's used by the USB library in it's cleanup things. An exception is thrown if it isn't there.

Reply:
In that case please can you add a comment explaining this?